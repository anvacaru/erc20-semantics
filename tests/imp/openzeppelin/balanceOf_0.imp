//when the requested account has no tokens
//it should return zero

int a, initialHolder, initialSupply, anotherAccount;
initialHolder  = 1;
initialSupply  = 100;
anotherAccount = 2;

makeThrowAsPrintTrue;
makeAccount(initialHolder, initialSupply);
makeCaller(initialHolder);
makeTotalSupply(initialSupply);
test.balanceOf(anotherAccount);