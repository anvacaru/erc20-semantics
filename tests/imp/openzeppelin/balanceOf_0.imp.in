//when the requested account has no tokens
//it should return zero

int a, initialHolder, initialSupply, anotherAccount;
initialHolder  = 1;
initialSupply  = 100;
anotherAccount = 2;

makeThrowAsPrintTrue;
makeAccount(initialHolder, initialSupply);
makeTotalSupply(initialSupply);
test.balanceOf(anotherAccount);