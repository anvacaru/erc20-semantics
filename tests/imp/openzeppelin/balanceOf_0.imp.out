Balance of 2 is <IMP>
  <ERC20>
    <caller>
      0
    </caller>
    <k>
      balanceOf ( 2 ) ~> #freezerprint(_)_IMP0_ ( ) ~> print ( "\n" ) ;
    </k>
    <accounts>
      <account>
        <id>
          1
        </id>
        <balance>
          100
        </balance>
      </account>
    </accounts>
    <allowances>
      <allowance1>
        <owner>
          1
        </owner>
        <spenders>
          .AllowCellBag
        </spenders>
      </allowance1>
    </allowances>
    <log>
      Events:
    </log>
    <supply>
      100
    </supply>
  </ERC20>
  <state>
    a |-> 2
    anotherAccount |-> 2
    initialHolder |-> 1
    initialSupply |-> 100
  </state>
  <input>
    ListItem ( #buffer ( "" ) )
    ListItem ( "on" )
    ListItem ( #istream ( 0 ) )
  </input>
  <output>
    ListItem ( #ostream ( 1 ) )
    ListItem ( "on" )
    ListItem ( #buffer ( "" ) )
  </output>
  <throwAsPrint>
    true
  </throwAsPrint>
  <printStrings>
    true
  </printStrings>
</IMP>
